clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -849,6 +850,9 @@ void EmitAssemblyHelper::CreatePasses(legacy::PassManager &MPM,
     FPM.add(createVerifierPass());
 
   // Set up the per-module pass manager.
+  if (CodeGenOpts.VerifyModule && LangOpts.SYCLIsDevice)
```

Comment:
Let's add this pass unconditionally, as it is not really a verification, it is semantic analysis normally done by FE